core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -94,199 +93,186 @@
-        }
+    /** See {@code EntrySet.releaseNewValue} for more information */
+    void releaseNewValue(ThreadContext ctx) {
+        entrySet.releaseNewValue(ctx);
     }
 
     /**
-     * release key in slice, currently not in use, waiting for GC to be arranged
-     *
-     * @param lookUp*/
-    void releaseKey(EntrySet.LookUp lookUp) {
-        entrySet.releaseKey(lookUp);
+     * @param key a key buffer to be updated with the minimal key
+     * @return    true if successful
+     */
+    boolean readMinKey(EntrySet.KeyBuffer key) {
+        return entrySet.readKey(key, entrySet.getHeadNextIndex());
     }
 
-    // Use this function to release an unreachable value reference
-    void releaseValue(EntrySet.OpData opData) {
-        entrySet.releaseValue(opData);
+    /**
+     * @param key a key buffer to be updated with the maximal key
+     * @return    true if successful
+     */
+    boolean readMaxKey(EntrySet.KeyBuffer key) {
+        return entrySet.readKey(key, getLastItemEntryIndex());
     }
 
-    // Check if value reference is valid. Doesn't check further than that
-    // (meaning whether the underlined off-heap is marked deleted or version is negative)
-    boolean isValueRefValid(int ei) {
-      return entrySet.isValueRefValid(ei);
+    /**
+     * @return the index of the first item in the chunk
+     * See {@code EntrySet.getHeadNextIndex} for more information.
+     */
+    final int getFirstItemEntryIndex() {
+        return entrySet.getHeadNextIndex();
     }
 
-    Slice buildValueSlice(int ei) {
-      return entrySet.buildValueSlice(ei);
+    /**
+     * Finds the last sorted entry.
+     * @return the last sorted entry
+     */
+    private int getLastItemEntryIndex() {
+        int sortedCount = this.sortedCount.get();
+        int entryIndex = sortedCount == 0 ? entrySet.getHeadNextIndex() : sortedCount;
+        int nextEntryIndex = entrySet.getNextEntryIndex(entryIndex);
+        while (nextEntryIndex != NONE_NEXT) {
+            entryIndex = nextEntryIndex;
+            nextEntryIndex = entrySet.getNextEntryIndex(entryIndex);
+        }
+        return entryIndex;
     }
 
-    EntrySet.LookUp buildLookUp(int ei) {
-        return entrySet.buildLookUp(ei);
-    }
 
-    ByteBuffer readKeyFromEntryIndex(int ei) {
-        return entrySet.readKey(ei);
-    }
+    /********************************************************************************************/
+    /*-----------------------  Methods for looking up item in this chunk -----------------------*/
 
-    void setRKeyBuffer(int ei, OakDetachedReadKeyBuffer keyRef) {
-        entrySet.setKeyOutputRBuff(ei, keyRef);
-    }
+    /**
+     * look up a key
+     *
+     * @param ctx the context that will follow the operation following this key lookup.
+     *            The ctx will describe the condition of the value associated with {@code key}.
+     *            If {@code (ctx.isKeyValid() && ctx.isValueValid()) == True}, the key and value was found.
```

Comment:
I don't think this is a correct place to explain all that. The check for key/value validity is not done in the lookUp() method. Or does it appear everywhere ctx is used?

Reply:
Again. This is the same exact comment from before this PR.
I agree that it might be redundant. I vote for the complete removal of this comment.
Let me know what you think.